packages/plugin-ext-vscode/src/node/plugin-vscode-init.ts

```diff
@@ -19,6 +19,7 @@
 import * as theia from '@theia/plugin';
 import { BackendInitializationFn, PluginAPIFactory, Plugin, emptyPlugin } from '@theia/plugin-ext';
 import { VSCODE_DEFAULT_API_VERSION } from '../common/plugin-vscode-types';
+import { realpathSync } from 'fs';
```

Comment:
Please do not use blocking fs call.

Reply:
The `doInitialization` function where `realpathSync` is being called is a synchronous function, as such we would want it to error out in case `realpathSync` fails for some reason. An async implementation can possibly lead to a situation where `realpathSync` will fail silently whereas the call to `doInitialization` would succeed

Reply:
I believe the sync vs async issue should be resolved together with the issue @martin-fleck-at brought up.

Reply:
@msujew Thanks for your comment! I've made the changes as suggested by @martin-fleck-at and am using sync calls for the `doInitialization` methods and async one for the `readPackage` method. Hope that resolves this